sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_set/feature_set_materialization_metadata.py

```diff
@@ -45,17 +45,30 @@ def __init__(
         self.tags = tags
 
     @classmethod
-    def _from_rest_object(cls, obj: RestFeaturesetJob) -> "FeatureSetMaterializationMetadata":
+    def _from_rest_object(cls, obj: RestJobBase) -> "FeatureSetMaterializationMetadata":
         if not obj:
             return None
+        job_properties = obj.properties
+        job_type = job_properties.properties.get("azureml.FeatureStoreJobType", None)
+        feature_window_start_time = job_properties.properties.get("azureml.FeatureWindowStart", None)
+        feature_window_end_time = job_properties.properties.get("azureml.FeatureWindowEnd", None)
+
+        time_format = "%Y-%m-%dT%H:%M:%SZ"
+        feature_window_start_time = (
+            datetime.strptime(feature_window_start_time, time_format) if feature_window_start_time else None
+        )
+        feature_window_end_time = (
+            datetime.strptime(feature_window_end_time, time_format) if feature_window_end_time else None
+        )
+
         return FeatureSetMaterializationMetadata(
-            type=FeaturestoreJobTypeMap.get(obj.type),
-            feature_window_start_time=obj.feature_window.feature_window_start,
-            feature_window_end_time=obj.feature_window.feature_window_end,
-            name=obj.job_id,
-            display_name=obj.display_name,
-            creation_context=SystemData(created_at=obj.created_date),
-            duration=obj.duration,
-            status=obj.status,
-            tags=obj.tags,
+            type=FeaturestoreJobTypeMap.get(job_type),
+            feature_window_start_time=feature_window_start_time,
+            feature_window_end_time=feature_window_end_time,
+            name=obj.name,
+            display_name=job_properties.display_name,
+            creation_context=SystemData(created_at=obj.system_data.created_at),
+            status=job_properties.status,
+            tags=job_properties.tags,
+            duration=None,
```

Comment:
Do we not have duration in the JobBase?

Reply:
Yes, there's no duration found in JobBase.